### PR TITLE
feat: embedded knowledge base with viewer, help panel, and search

### LINKED ISSUE #11: Embedded knowledge base
## Summary

Build the embedded documentation viewer that serves as both app help and a comprehensive datacenter design reference. Every concept in the UI links to relevant documentation via a contextual slide-out panel. The knowledge base ships with the app, works offline, and aims to become the definitive open reference on datacenter infrastructure design.

## Features / Scope

| Feature | Description |
|---------|-------------|
| Knowledge viewer | Full-page viewer at `/knowledge` with hierarchical TOC navigation |
| Contextual help panel | Slide-out panel triggered by help buttons throughout the UI, with popout option to open in full viewer |
| Markdown rendering | Render articles with support for standard Markdown, Mermaid diagrams, math notation (KaTeX/MathJax), and images |
| Full-text search | Search across all knowledge base articles with highlighted results |
| Article hierarchy | Articles organized in categories (e.g., `networking/`, `power/`, `rack-design/`) |
| Initial content | Seed articles: Clos fabric fundamentals, oversubscription, ECMP, rack design basics, power planning |
| Knowledge API | Go backend serves article content and metadata via REST |

## Architecture / Design Notes

- **Module location**: Frontend component in `frontend/src/app/features/knowledge/`, shared help button component in `frontend/src/app/shared/`
- **Storage**: Markdown files in `docs/knowledge/`, organized hierarchically (e.g., `networking/clos-fabrics.md`, `power/rack-power-budgets.md`)
- **Content serving**: Go backend reads `docs/knowledge/` at startup, builds an index (title, path, category, headings), and serves via API. In production, content is embedded in the Go binary.
- **Article format**: Markdown with YAML frontmatter for metadata:
  ```yaml
  ---
  title: Clos Fabric Fundamentals
  category: networking
  tags: [clos, fabric, spine-leaf, topology]
  ---
  ```
- **Linking convention**: Other features reference articles by path slug. A shared `HelpButtonComponent` accepts an article path and opens the slide-out panel:
  ```html
  <app-help-button article="networking/oversubscription"></app-help-button>
  ```
- **Rendering pipeline**: Markdown → HTML with plugins for Mermaid (diagram rendering), KaTeX (math notation), and syntax highlighting
- **Search**: Client-side full-text search using a pre-built index (generated at build time or served via API). No server-side search needed for the initial article set.

## Integration Points

- **Shared `HelpButtonComponent`**: Lives in `frontend/src/app/shared/` — imported by every feature module that needs contextual help (#3 rack modeling, #4 fabric designer, #7 capacity tracking, #9 metrics dashboard)
- **App shell (#12)**: Knowledge base is a top-level route in the sidebar navigation
- **Slide-out panel**: A singleton service in `core/` manages panel state so any feature can trigger it
- **Article content**: Authors add `.md` files to `docs/knowledge/` — no frontend changes needed for new articles
- **Go backend**: Knowledge API endpoints added to `server/internal/api/handlers/`

## Acceptance Criteria

- [ ] Knowledge viewer component in `frontend/src/app/features/knowledge/` with full-page route at `/knowledge`
- [ ] Slide-out contextual help panel triggered by `HelpButtonComponent`, with popout button to navigate to full viewer
- [ ] Render Markdown with Mermaid diagram support, KaTeX math notation, images, and syntax highlighting
- [ ] Full-text search across all articles with result highlighting
- [ ] Hierarchical table of contents with category grouping
- [ ] YAML frontmatter parsed for title, category, and tags
- [ ] Internal article cross-links resolve correctly (e.g., `[oversubscription](networking/oversubscription)`)
- [ ] `HelpButtonComponent` in `shared/` accepts an article path and opens the panel
- [ ] REST API: `GET /api/knowledge` (article index), `GET /api/knowledge/:path` (article content)
- [ ] Initial articles written: Clos fabric fundamentals, oversubscription ratios, ECMP, rack design basics, power planning
- [ ] Works fully offline (content embedded in the build)
- [ ] Unit tests for Markdown rendering (including Mermaid and KaTeX blocks)
- [ ] Unit tests for search indexing and querying
- [ ] Unit tests for help button component and panel service
- [ ] Unit tests for knowledge API handlers
- [ ] E2e test: navigate to knowledge base, search, open an article, verify rendering
- [ ] Tests pass, build succeeds

## Edge Cases

- Broken article link from a help button → panel shows "Article not found" with suggestion to search
- Search with no results → friendly empty state with suggestions
- Very long articles → scroll position preserved when switching between panel and full viewer
- Internal cross-links to non-existent articles → render as plain text with console warning in dev mode
- Mermaid diagram syntax error → show error message inline instead of blank space
- Math notation syntax error → render raw LaTeX as fallback
- Articles with no frontmatter → use filename as title, "uncategorized" as category

## Documentation

- [ ] Article authoring guide in `docs/knowledge/README.md` (frontmatter format, supported Markdown extensions, image handling, how to add a new article)
- [ ] Linking convention documented in `CLAUDE.md` (`HelpButtonComponent` usage, article path scheme)
- [ ] Knowledge base architecture in `CLAUDE.md` (content pipeline, search approach)

## Dependencies

None — can be built in parallel with other features. The `HelpButtonComponent` will be consumed by other features as they are built.

### PR DESCRIPTION
**Pull request by rnwolfe**
## Summary

- 5 Markdown articles in `docs/knowledge/` with YAML frontmatter covering Clos fabric, oversubscription, ECMP, rack design, and power planning
- Go REST API: `GET /api/knowledge` (index) and `GET /api/knowledge/{path...}` (full article) — content embedded in binary at build time via Makefile copy step
- Angular `KnowledgePanelService` (core singleton) manages slide-out panel state via signals
- `KnowledgeViewerComponent` at `/knowledge` with hierarchical TOC, category grouping, and full-text client-side search
- `ArticleViewComponent` renders Markdown with `marked` (syntax highlighting via `highlight.js`), Mermaid diagrams, and KaTeX math
- `KnowledgePanelComponent`: slide-out panel with animation, popout-to-full-viewer button
- `HelpButtonComponent` in `shared/`: `<app-help-button article="networking/oversubscription">` opens slide-out panel with toggle behaviour

## Acceptance Criteria

- [x] Knowledge viewer component in `frontend/src/app/features/knowledge/` at `/knowledge`
- [x] Slide-out panel with `HelpButtonComponent`, popout to full viewer
- [x] Markdown rendering with Mermaid, KaTeX math, images, syntax highlighting
- [x] Full-text search with result highlighting (client-side, score-ranked)
- [x] Hierarchical TOC with category grouping
- [x] YAML frontmatter parsed (title, category, tags)
- [x] Internal cross-links resolve correctly (via `data-knowledge-link` attribute + click handler)
- [x] `HelpButtonComponent` in `shared/` opens the panel
- [x] REST API: `GET /api/knowledge`, `GET /api/knowledge/:path`
- [x] 5 initial articles written in `docs/knowledge/`
- [x] Works offline (embedded content via `go:embed` at build time)
- [x] Unit tests for rendering, search, help button, panel service, API handlers
- [x] E2e test: `frontend/e2e/knowledge.spec.ts` (runs against real backend)
- [x] Tests pass (8 test files, 40 tests), build succeeds, lint clean

## Test plan

- [x] `make test-server` — 40 Go tests pass (includes knowledge package and handlers)
- [x] `make test-frontend` — 40 Angular tests pass across 8 spec files
- [x] `make lint` — all files pass linting (Go vet + ESLint)
- [x] `make build` — server binary and Angular bundle build successfully
- [ ] `make test-e2e` — requires running server (Playwright, verified spec exists)

Closes #11

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/internal/knowledge/knowledge.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+		switch key {
+		case "title":
+			a.Title = unquote(val)
+		case "category":
+			a.Category = unquote(val)
+		case "tags":
+			a.Tags = parseTags(val)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (rnwolfe) [reply server/internal/knowledge/knowledge.go]: Fixed in PR #18: `parseFrontmatter` now checks `scanner.Err()` after the loop and surfaces the error.

File: server/internal/knowledge/knowledge.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	s = strings.TrimSpace(s)
+	s = strings.TrimPrefix(s, "[")
+	s = strings.TrimSuffix(s, "]")
+	parts := strings.Split(s, ",")
+	var tags []string
+	for _, p := range parts {
+		t := strings.TrimSpace(p)
+		if t != "" {
+			tags = append(tags, t)
+		}
```
[the remark this answers is not in the document]
Comment (rnwolfe) [reply server/internal/knowledge/knowledge.go]: Fixed in PR #18: `parseTags` now applies `strings.Trim(t, "'\"")` to each tag, so quoted YAML items like `["clos", "fabric"]` produce clean strings.
